src/Main.elm

```diff
@@ -16,25 +17,106 @@ main =
 
 
 type alias Model =
-    {}
+    { display : String
+    , displayStatus : DisplayStatus
+    , decimalStatus : DecimalStatus
+    }
 
 
 init : Model
 init =
-    {}
+    { display = "0"
+    , displayStatus = Locked
+    , decimalStatus = NoDecimal
+    }
 
 
 
 -- Update
 
 
 type Msg
-    = Message
+    = UpdateDisplay String
+
+
+type InputType
+    = Num
+    | Decimal
+
+
+type DisplayStatus
+    = Locked
+    | Unlocked
+
+
+type DecimalStatus
+    = YesDecimal
+    | NoDecimal
+
+
+inputType : String -> InputType
+inputType string =
+    if String.contains "." string then
+        Decimal
+
+    else
+        Num
```

Comment:
Instead of having a function determine the input type from a string, I would use a [custom type](https://guide.elm-lang.org/types/custom_types.html) here.

You can avoid several things this way:
* You don't have to deal with the question of what to do if someone provides a malformed string, like `"a"`. Right now this function would claim that `"a"` is a number.
* The `updateDisplay` function could be changed to _only allow_ specific characters if desired. As it is right now, you could accidentally write code that updates the display to display `"foobar"`.